src/connect.ts

```diff
@@ -1,10 +1,59 @@
+import {
+  Manifest,
+  FlashState,
+  FlashError,
+  FlashStateType,
+} from "./const.js";
 import type { InstallButton } from "./install-button.js";
 import "./install-dialog.js";
```

Comment:
Let's dynamically import this, while we're fetching the manifest to speed up opening the request port dialog.

Reply:
They're needed before fetching routine as FlashState is passed to call connect.

Reply:
@balloob If you can clarify this, Then I can implement the changes you've asked for.